Backend/app/routes/brand_dashboard.py

```
@@ -0,0 +1,1132 @@
+    campaign_update: dict,
+    campaign_id: str = Path(..., min_length=36, max_length=36, regex=r"^[a-fA-F0-9\-]{36}$", description="Campaign ID (UUID)"),
+    brand_id: str = Query(..., min_length=36, max_length=36, regex=r"^[a-fA-F0-9\-]{36}$", description="Brand user ID (UUID)")
+):
+    """
+    Update campaign details
+    """
+    try:
+        # Verify campaign belongs to brand
+        existing = supabase.table("sponsorships").select("*").eq("id", campaign_id).eq("brand_id", brand_id).execute()
+        if not existing.data:
+            raise HTTPException(status_code=404, detail="Campaign not found")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)